Add template information to search application API docs (#95591)
elastic/elasticsearch: docs/reference/search-application/apis/put-search-application.asciidoc

[[put-search-application-path-params]]
==== {api-path-parms-title}

`create`::
(Optional, Boolean) If `true`, this request cannot replace or update existing Search Applications.
Defaults to `false`.

`<body>`::
(Optional, object)
Contains parameters for a search application:
+
.Properties of `<body>` objects
[%collapsible%open]
====
`indices`::
(Required, array of strings)
The <<indices,indices>> associated with this search application. All indices need to exist in order to be added to a search application.

> `analytics_collection_name`::

Reviewer: I think we should have removed this param, at least from the requests / responses given that we're not using it - and it's confusing for users to have something that has no practical purpose.

Author: You are absolutely correct - I didn't want to remove it until @aznick confirmed, but he did last night and I'm removing it now 😁

(Optional, string) The name of the behavioral analytics association being used with this search application.

> As of this release, this is used for informational purposes only. For more information see: <<behavioral-analytics-apis

Reviewer 2: ```suggestion
Presently, this is used for informational purposes only. For more information see: <<behavioral-analytics-apis
```
If this text carries over to the next release, the original text may no longer make sense.

`template`::
(Optional, object)

> The <<search-template>> associated with this search application. The search application's template is only stored and accessible through the search application.

Reviewer 2: ```suggestion
The <<search-template,search template>> associated with this search application. The search application's template is only stored and accessible through the search application.
```

When using a link within a sentence, I suggest always providing explicit link text. Several times I've seen the name of a page change, and the new implicit link text breaks the sentence.

I only marked this here, but it would apply throughout this PR.

- This search template must be a Mustache template.
- The template must contain a Mustache script and script source.
- The template may be modified with subsequent <<put-search-application>> requests.

> - If no template is specified when creating a search application, or if a template is removed from a search application, we use the default <<query-string-query-ex-request>> defined in the template examples as a default.

Reviewer 2: Can revise to avoid "default" twice.

> - This template will be used by the <<search-application-search,search application search>> API to execute searches.

Reviewer: I don't think search API is documented. Will that be done as part of https://github.com/elastic/enterprise-search-team/issues/3452, or should we create a separate issue?

> - This template will be used by the <<search-application-search,search application search>> API to execute searches.

Reviewer: I don't think search API is documented. Will that be done as part of https://github.com/elastic/enterprise-search-team/issues/3452, or should we create a separate issue?

Author: I added documentation for the search API call in a previous PR, but it's not currently part of the changes in this PR.  [Link](https://github.com/elastic/elasticsearch/blob/main/docs/reference/search-application/apis/search-application-search.asciidoc). If you recommend any changes I'm happy to add them to this PR.

Author: Oh shoot, you're right, this might not be in 8.8.

Author: Let's get this PR merged and I'll create a separate 8.8 PR for the search docs.
